aten/src/ATen/native/mkldnn/Conv.cpp

```diff
@@ -124,6 +141,81 @@ Tensor mkldnn_convolution(
   }
 }
 
+Tensor mkldnn_convolution_relu(
+    const Tensor& input,
+    const Tensor& weight,
+    const c10::optional<Tensor>& bias_opt,
+    IntArrayRef stride,
+    IntArrayRef padding,
+    IntArrayRef dilation,
+    int64_t groups) {
+  TORCH_CHECK(input.dim() == 4 || input.dim() == 5,
+      "mkldnn_convolution_relu: currently mkldnn only support 2d or 3d convolution relu fusion");
+  TORCH_CHECK(input.scalar_type() == ScalarType::Float || input.scalar_type() == ScalarType::BFloat16,
+      "mkldnn_convolution_relu: currently mkldnn only support float or bfloat16 input");
+
+  // See [Note: hacky wrapper removal for optional tensor]
+  c10::MaybeOwned<Tensor> bias_maybe_owned = at::borrow_from_optional_tensor(bias_opt);
+  const Tensor& bias = *bias_maybe_owned;
+
+  if (input.size(0) == 0) {
+    // don't failed when given a empty input for jit fusion path.
+    std::vector<int64_t> o;
+    auto dim = weight.ndimension() - 2;
+    auto stride_expanded = expand_param_if_needed(stride, "stride", dim);
+    auto padding_expanded = expand_param_if_needed(padding, "padding", dim);
+    auto dilation_expanded = expand_param_if_needed(dilation, "dilation", dim);
+    o = conv_output_size(input.sizes(), weight.sizes(), padding_expanded,
+                         stride_expanded, dilation_expanded);
```

Comment:
@jbschlosser this sucks, I wonder if convolution can be made structured after you do your work

Reply:
True, I don't see why not, but then I'm not at all familiar with the limitations of structured